src/main/resources/default/assets/javascript/client-pagination.js

```diff
@@ -6,47 +6,60 @@
  * http://www.scireum.de - info@scireum.de
  */
 
-function paginate(paginationControls, update) {
-    const pagination = {
-        previousBtn: $("<li><a href='#' aria-label='Previous'><span aria-hidden='true'>&laquo;</span></a></li>"),
-        pageIndicator: $("<li><a href='#' aria-label='Page'/></li>"),
-        nextBtn: $("<li><a href='#' aria-label='Next'><span aria-hidden='true'>&raquo;</span></a></li>"),
-        currentPage: 1
-    }
-
-    pagination.reset = function () {
-        pagination.updatePage(1)
-    };
-
-    pagination.updatePage = function (page) {
-        pagination.pageIndicator.children("a").text(page);
-        pagination.currentPage = page;
-        update(page, pagination);
-    }
-
-    pagination.next = function () {
+function Pagination(paginationContainer, pageSize, update) {
+    const self = this;
+
+    paginationContainer.innerHTML = "";
+
+    const paginationList = document.createElement("ul");
+    paginationList.classList.add("pagination")
+    paginationContainer.appendChild(paginationList);
+
+    this.previousBtn = document.createElement("li");
+    this.previousBtn.innerHTML = "<a href='#' aria-label='Previous'><span aria-hidden='true'>&#8592;</span></a>";
+    this.nextBtn = document.createElement("li");
+    this.nextBtn.innerHTML = "<a href='#' aria-label='Next'><span aria-hidden='true'>&#8594;</span></a>";
+    this.pageIndicator = document.createElement("li");
+    this.pageIndicator.innerHTML = "<a href='#' aria-label='Page'/>";
```

Comment:
Ist nur ein Codestyle Vorschlag an den ich mich selbst halte, der aber nicht unbedingt von allen umgesetzt werden muss:

Ich prefixe Variablen die ein DOM Element halten immer mit _ und solche die normale Datenstrukturen halten versehe ich mit keinem Prefix. So sieht man auf einen Blick wann am DOM geschraubt wird und wann nicht.

Reply:
Ist ergänzt. Können/Sollen/Wollen wir das in Styleguide aufnehmen?